File: solutions/zig/01-ry8/code/src/main.zig
const std = @import("std");
const stdout = std.fs.File.stdout();

pub fn main() !void {
    const args = try std.process.argsAlloc(std.heap.page_allocator);
    defer std.process.argsFree(std.heap.page_allocator, args);

    if (args.len < 3) {
        try stdout.writeAll("Usage: ./your_program.sh tokenize <filename>\n");
        std.process.exit(1);
    }

    const command = args[1];
    const filename = args[2];

    if (!std.mem.eql(u8, command, "tokenize")) {
        try stdout.writeAll("Unknown command: {s}\n");
        std.process.exit(1);
    }

    const file_contents = try std.fs.cwd().readFileAlloc(std.heap.page_allocator, filename, std.math.maxInt(usize));
    defer std.heap.page_allocator.free(file_contents);

    if (file_contents.len > 0) {
        @panic("Scanner not implemented");
    } else {
        try stdout.writeAll("EOF  null\n"); // Placeholder, replace this line when implementing the scanner
    }
}

File: solutions/typescript/01-ry8/code/app/main.ts
import fs from "fs";

const args: string[] = process.argv.slice(2); // Skip the first two arguments (node path and script path)

if (args.length < 2) {
  console.error("Usage: ./your_program.sh tokenize <filename>");
  process.exit(1);
}

const command: string = args[0];

if (command !== "tokenize") {
  console.error(`Usage: Unknown command: ${command}`);
  process.exit(1);
}

const filename: string = args[1];

const fileContent: string = fs.readFileSync(filename, "utf8");

if (fileContent.length !== 0) {
  throw new Error("Scanner not implemented");
} else {
  console.log("EOF  null");
}
